arangod/RocksDBEngine/RocksDBZkdIndex.cpp

```diff
@@ -691,6 +694,11 @@ auto extractAttributeValues(
     if (s.isNone()) {
       s = VPackSlice::nullSlice();
     }
+
+    if (s.isNull() && !nullAllowed) {
+      return {TRI_ERROR_ARANGO_DOCUMENT_KEY_MISSING};
```

Comment:
Is this the best error code we can use here? In other cases we are using it to indicate that the `_key` attribute is missing.
This is not the case here. I am happy if we use this error code here too, but it should be a conscious decision.

Reply:
The code is only used internally and never leaves the database. It's only used within this file. Happy to change it, I just don't want to add a new error code here. Maybe something more generic?

Reply:
If it is used only internally, I think it is fine as is. Would be useful to add a code comment somewhere to indicate that this is only an internal code to signal a certain condition.

Reply:
I will add that comment in a follow up PR.